addOns/ascanrulesAlpha/src/main/java/org/zaproxy/zap/extension/ascanrulesAlpha/MongoDbInjection.java

```diff
@@ -0,0 +1,687 @@
+            SLEEP_SHORT_TIME = 1000;
+            SLEEP_LONG_TIME = 3000;
+            doCrashScan = true;
+            doAllDataScan = true;
+            doTimedScan = true;
+            doJsonScan = true;
+            getMoreConfidence = true;
+            doAuthBypass = true;
+        }
+    }
+
+    @Override
+    public void scan(HttpMessage msg, NameValuePair originalParam) {
+        isJsonPayload = originalParam.getType() == NameValuePair.TYPE_JSON;
+        super.scan(msg, originalParam);
+    }
+
+    @Override
+    public void scan(HttpMessage msg, String param, String value) {
+        boolean isBingo = false;
+        List<Integer> rtt = new ArrayList<>();
+        HttpMessage msgInjAttack, counterProofMsg;
+        String bodyInjAttack,
+                bodyCounterProof,
+                bodyBase = getBaseMsg().getResponseBody().toString();
+
+        if (!inScope(getTech())) {
+            return;
+        }
+        if (LOG.isDebugEnabled()) {
+            LOG.debug(
+                    "\nScannning URL ["
+                            + msg.getRequestHeader().getMethod()
+                            + "] ["
+                            + msg.getRequestHeader().getURI()
+                            + "] on param: ["
+                            + param
+                            + "] with value: ["
+                            + value
+                            + "] for MongoDB Injection");
+        }
+        // injection attack to url-encoded query parameters
+        if (doAllDataScan) {
+            if (LOG.isDebugEnabled()) {
+                LOG.debug(
+                        "\nStarting with the url-encoded query injection package of attack rules:");
+            }
+            int index = 0;
+            for (String valueInj : ALL_DATA_VALUE_INJECTION) {
+                String paramInj = param + ALL_DATA_PARAM_INJECTION[index++];
+                if (isStop()) {
+                    if (LOG.isDebugEnabled()) {
+                        LOG.debug(STOP_LOG);
+                    }
+                    return;
+                }
+                if (LOG.isDebugEnabled()) {
+                    LOG.debug("\nTrying with the value: " + paramInj + valueInj);
+                }
+                try {
+                    msgInjAttack = getNewMsg();
+                    setParameter(msgInjAttack, paramInj, valueInj);
+                    sendAndReceive(msgInjAttack, false);
+                    rtt.add(msgInjAttack.getTimeElapsedMillis());
+                    bodyInjAttack = msgInjAttack.getResponseBody().toString();
+                    if (msgInjAttack.getResponseHeader().getStatusCode() != HttpStatusCode.OK) {
+                        continue;
+                    }
+                    if (!bodyBase.equals(bodyInjAttack)) {
+                        counterProofMsg = getNewMsg();
+                        setParameter(counterProofMsg, param + ZAP_QUERYSTRING, valueInj);
```

Comment:
I'd expect this to use a neutral check (e.g. `setParameter(counterProofMsg, param + "[$eq]", value);`), otherwise it might not work properly (if an error occurs with `$ZAP`).